File: backend/src/main/java/com/example/backend/models/UserProfile.java
package com.example.backend.models;

import jakarta.persistence.*;

import java.util.Arrays;

@Entity
public class UserProfile {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;

    private String userName;
    private String firstName;
    private String addition;
    private String lastName;
    private String bio;
    private String phoneNumber;
    private String residence;
    private String country;
    private String address;
    private String houseNumber;
    private String postalCode;

    @Lob
    @Column(name = "image_data", columnDefinition = "MEDIUMBLOB")
    private byte[] imageData;

    private String kvkNumber;
    private String linkedInUrl;

    public UserProfile() {
    }

    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public String getUserName() {
        return userName;
    }

    public void setUserName(String userName) {
        this.userName = userName;
    }

    public String getFirstName() {
        return firstName;
    }

    public void setFirstName(String firstName) {
        this.firstName = firstName;
    }

    public String getAddition() {
        return addition;
    }

    public void setAddition(String addition) {
        this.addition = addition;
    }

    public String getLastName() {
        return lastName;
    }

    public void setLastName(String lastName) {
        this.lastName = lastName;
    }

    public String getBio() {
        return bio;
    }

    public void setBio(String bio) {
        this.bio = bio;
    }

    public String getPhoneNumber() {
        return phoneNumber;
    }

    public void setPhoneNumber(String phoneNumber) {
        this.phoneNumber = phoneNumber;
    }

    public String getResidence() {
        return residence;
    }

    public void setResidence(String residence) {
        this.residence = residence;
    }

    public String getCountry() {
        return country;
    }

    public void setCountry(String country) {
        this.country = country;
    }

    public String getAddress() {
        return address;
    }

    public void setAddress(String address) {
        this.address = address;
    }

    public String getHouseNumber() {
        return houseNumber;
    }

    public void setHouseNumber(String houseNumber) {
        this.houseNumber = houseNumber;
    }

    public String getPostalCode() {
        return postalCode;
    }

    public void setPostalCode(String postalCode) {
        this.postalCode = postalCode;
    }

    public byte[] getImageData() {
        return imageData;
    }

    public void setImageData(byte[] imageData) {
        this.imageData = imageData;
    }

    public String getKvkNumber() {
        return kvkNumber;
    }

    public void setKvkNumber(String kvkNumber) {
        this.kvkNumber = kvkNumber;
    }

    public String getLinkedInUrl() {
        return linkedInUrl;
    }

    public void setLinkedInUrl(String linkedInUrl) {
        this.linkedInUrl = linkedInUrl;
    }

    @Override
    public String toString() {
        return "UserProfile{" +
                "id=" + id +
                ", userName='" + userName + '\'' +
                ", firstName='" + firstName + '\'' +
                ", addition='" + addition + '\'' +
                ", lastName='" + lastName + '\'' +
                ", bio='" + bio + '\'' +
                ", phoneNumber='" + phoneNumber + '\'' +
                ", residence='" + residence + '\'' +
                ", country='" + country + '\'' +
                ", address='" + address + '\'' +
                ", houseNumber='" + houseNumber + '\'' +
                ", postalCode='" + postalCode + '\'' +
                ", imageData=" + Arrays.toString(imageData) +
                ", kvkNumber='" + kvkNumber + '\'' +
                ", linkedInUrl='" + linkedInUrl + '\'' +
                '}';
    }
}

File: frontend/src/models/UserProfile.js
export class UserProfile {
    constructor(id,
                firstName,
                addition,
                bio,
                address,
                houseNumber,
                postalCode,
                phoneNumber,
                country,
                residence,
                kvkNumber,
                linkedInUrl,
                imageData
                ) {
        this.id = id;
        this.firstName = firstName;
        this.addition = addition;
        this.bio = bio;
        this.address = address;
        this.houseNumber = houseNumber;
        this.postalCode = postalCode;
        this.phoneNumber = phoneNumber;
        this.country = country;
        this.residence = residence;
        this.kvkNumber = kvkNumber;
        this.linkedInUrl = linkedInUrl;
        this.imageData = imageData;
    }

    static create(pId = 0, pFirstName, pAddition, pBio, pAddress, pHouseNumber, pPostalCode, pPhoneNumber, pCountry, pResidence, pKvkNumber, plinkedInUrl, pImageData, pEmail, pRole) {
        return new UserProfile(
            pId,
            pFirstName,
            pAddition,
            pBio,
            pAddress,
            pHouseNumber,
            pPostalCode,
            pPhoneNumber,
            pCountry,
            pResidence,
            pKvkNumber,
            plinkedInUrl,
            pImageData,
            pEmail,
            pRole
        );
    }


    static copyConstructor(userProfile) {
        if (userProfile == null) {
            return null;
        }
        return Object.assign(new UserProfile(), userProfile);
    }
}
